### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by nabobalis**
Trying to work out edge cases.

### CODE REVIEW

File: tests/test_versions.py [hunk 1]
```diff
@@ -16,14 +17,22 @@ def test_tighter_lower_bound_any():
...
-    lower_bound = Version("3.8.0")
+    spec = SpecifierSet(">=1.0,!= 1.3.4.*,< 2.0")
+    lower_bound = Version("1.4.0")
     tightened = tighten_lower_bound(spec, lower_bound)
-    assert tightened == SpecifierSet("~= 0.9,>=3.8.0,!=1.3.4.*,<2.0")
+    assert tightened == SpecifierSet(">=1.4.0,!=1.3.4.*,<2.0")
 
 
 def test_tighter_lower_bound_adds_lower_bound_if_not_present():
-    spec = SpecifierSet("~=0.9,!=1.3.4.*,<2.0")
```
Comment (nabobalis) [inline tests/test_versions.py:26]: I need to keep a test like this.
